tools/icqsol_translate_shape/icqsol_translate_shape.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<tool id="icqsol_translate_shape" name="translate shape" version="1.0.0">
+    <description></description>
+    <macros>
+        <import>icqsol_macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command>
+<![CDATA[
+    python $__tool_directory__/icqsol_translate_shape.py
+    --input "$input"
+    --displacement_x $displacement_x
+    --displacement_y $displacement_y
+    --displacement_z $displacement_z
+    --output "$output"
+    --output_vtk_type $output_vtk_type
+]]>
+    </command>
+    <inputs>
+        <param name="input" type="data" format="plyascii,plybinary,vtkascii,vtkbinary" label="Shape">
+            <validator type="dataset_ok_validator" />
+        </param>
+        <param name="displacement_x" type="float" value="0" label="X coordinate of displacement" help="Floating point number.">
+        </param>
+        <param name="displacement_y" type="float" value="0" label="Y coordinate of displacement" help="Floating point number.">
+        </param>
+        <param name="displacement_z" type="float" value="0" label="Z coordinate of displacement" help="Floating point number.">
+        </param>
+        <expand macro="output_vtk_type_params" />
+    </inputs>
+    <outputs>
+        <data name="output" format="vtkascii" label="${tool.name} ${on_string} Max edge length: ${max_edge_length}">
+            <actions>
+                <action type="format">
+                    <option type="from_param" name="output_vtk_type" />
+                </action>
+            </actions>
+        </data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" value="box_with_surface_field.vtkascii" ftype="vtkascii" />
+            <param name="displacement_x" value="-0.2" />
```

Comment:
You'll need these here:
            <param name="input_file_format_and_type" value="vtkascii" />
            <param name="input_dataset_type" value="POLYDATA" />